cairo-libs/packages/ics23/src/utils.cairo

```diff
@@ -216,3 +215,68 @@ pub fn array_u32_to_byte_array(array: Array<u32>) -> ByteArray {
     array_u8_to_byte_array(@array_u32_into_array_u8(array, 0, 0))
 }
 
+
+// ---------------------------------------------------------------
+// Implementation of partial ordering for `Array<u8>`
+// ---------------------------------------------------------------
+
+pub impl ArrayU8PartialOrd of PartialOrd<Array<u8>> {
+    fn le(lhs: Array<u8>, rhs: Array<u8>) -> bool {
+        lexicographical_cmp(lhs, rhs) != Ordering::Greater
+    }
+    fn ge(lhs: Array<u8>, rhs: Array<u8>) -> bool {
+        lexicographical_cmp(lhs, rhs) != Ordering::Less
+    }
+    fn lt(lhs: Array<u8>, rhs: Array<u8>) -> bool {
+        lexicographical_cmp(lhs, rhs) == Ordering::Less
+    }
+    fn gt(lhs: Array<u8>, rhs: Array<u8>) -> bool {
+        lexicographical_cmp(lhs, rhs) == Ordering::Greater
+    }
+}
+
+#[derive(Drop, Debug, PartialEq)]
+pub enum Ordering {
+    Equal,
+    Less,
+    Greater,
+}
+
+/// Lexicographical comparison of two `u8` arrays.
+pub fn lexicographical_cmp(lhs: Array<u8>, rhs: Array<u8>) -> Ordering {
+    let lhs_len = lhs.len();
+    let rhs_len = rhs.len();
+    let min_len = if lhs_len < rhs_len {
+        lhs_len
+    } else {
+        rhs_len
+    };
```

Comment:
I think `core::cmp::min` is available.

Reply:
[b32d64f](https://github.com/informalsystems/ibc-starknet/pull/305/commits/b32d64f9dc447d84accf6e3eccfed36e4c03cc58)